File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/04.For-Loop/02-exercise/04.cleverLily.js
function cleverLily(input) {

    const lilysAge = Number(input[0]);
    const washingMachinePrice = Number(input[1]);
    const unitPriceOfAToy = Number(input[2]);

    let toysCount = 0;
    let savedMoney = 0;
    let giftMoney = 10;

    for (let i = 1; i <= lilysAge; i++) {
        if (i % 2 !== 0) {
            toysCount++;
        } else {
            savedMoney += giftMoney - 1;
            giftMoney += 10;
        }
    }

    const sumOfToys = toysCount * unitPriceOfAToy;
    const difference = Math.abs(sumOfToys + savedMoney - washingMachinePrice).toFixed(2);

    if (sumOfToys + savedMoney >= washingMachinePrice) {
        console.log(`Yes! ${difference}`);
    } else {
        console.log(`No! ${difference}`);
    }
}
cleverLily(["10", "170", "6"]);

File: Python_Basics/Homeworks-And-Labs/04.for_loop/02.exercise/04.clever_lily.py
age = int(input())
washing_machine_price = float(input())
unit_price_of_a_toy = int(input())

toys_count = 0
saved_money = 0
gift_money = 10

for i in range(1, age + 1):
    if i % 2 != 0:
        toys_count += 1
    else:
        saved_money += gift_money - 1
        gift_money += 10

sum_of_toys = toys_count * unit_price_of_a_toy
difference = abs(sum_of_toys + saved_money - washing_machine_price)

if sum_of_toys + saved_money >= washing_machine_price:
    print(f'Yes! {difference:.2f}')
else:
    print(f'No! {difference:.2f}')
